src/operator/tensor/elemwise_binary_broadcast_op-inl.cuh

```diff
@@ -58,8 +58,12 @@ BinaryBroadcastBackwardUseNone(const nnvm::NodeAttrs& attrs,
             ctx.requested[0].get_space_typed<gpu, 1, char>(
                 Shape1(workspace_size * sizeof(index_t)), s);
         if (out.shape_.Size() != 0) {
-          Reduce<red::sum, NDim, DType, LOP>(s, lhs, req[0], workspace, out);
-          Reduce<red::sum, NDim, DType, ROP>(s, rhs, req[1], workspace, out);
+          if (common::is_float(lhs.type_flag_)) {
+            Reduce<red::sum, NDim, DType, LOP>(s, lhs, req[0], workspace, out);
+          }
+          if (common::is_float(rhs.type_flag_)) {
+            Reduce<red::sum, NDim, DType, ROP>(s, rhs, req[1], workspace, out);
```

Comment:
Is it possible that `rhs.dtype` != DType?

Reply:
you're right, we need extra handling for that case.